core/domain/classifier_domain_test.py

```diff
@@ -269,3 +269,59 @@ def test_validation(self):
         with self.assertRaisesRegexp(utils.ValidationError, (
             'Expected training_data to be a list')):
             training_job.validate()
+
+
+class ClassifierExplorationMappingDomainTests(test_utils.GenericTestBase):
+    """Tests for the ClassifierExplorationMapping domain."""
+
+    def _get_mapping_from_dict(self, mapping_dict):
+        mapping = classifier_domain.ClassifierExplorationMapping(
+            mapping_dict['exp_id'],
+            mapping_dict['exp_version'],
+            mapping_dict['state_name'],
+            mapping_dict['classifier_id'])
+
+        return mapping
+
+    def test_to_dict(self):
+        expected_mapping_dict = {
+            'exp_id': 'exp_id1',
+            'exp_version': 2,
+            'state_name': 'state_name1',
+            'classifier_id': 'classifier_id1'
+        }
+        observed_mapping = self._get_mapping_from_dict(
+            expected_mapping_dict)
+        self.assertDictEqual(expected_mapping_dict,
+                             observed_mapping.to_dict())
+
+    def test_validation(self):
```

Comment:
@seanlip . I think we still have the wrong idea here. We are never working with unicode strings actually. When we create the storage model, any unicode string is converted to utf-8 (and possibly, I don't think we need to convert it back ever). So, we are completely working with only strings since unicode gets converted. So, basically, we should be able to assume that, in the logic layer, state names are always string (not unicode). Are there any issues here?